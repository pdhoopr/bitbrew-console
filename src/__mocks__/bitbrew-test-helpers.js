import diff from 'jest-diff';
import ShallowRenderer from 'react-test-renderer/shallow';

export function diffComponents(baseComponent, compareComponent) {
  const changes = diff(baseComponent, compareComponent, {
    aAnnotation: `<BaseComponent />`,
    bAnnotation: `<CompareComponent />`,
    contextLines: 3,
    expand: false,
  });
  return `Diff:\n${changes}`;
}

export function shallow(component) {
  const renderer = new ShallowRenderer();
  renderer.render(component);
  return renderer.getRenderOutput();
}

export const fixtures = {
  accessToken:
    'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJPbmxpbmUgSldUIEJ1aWxkZXIiLCJpYXQiOjE1MjUxMTUzNTAsImV4cCI6MTU1NjY1MTM1MCwiYXVkIjoid3d3LmV4YW1wbGUuY29tIiwic3ViIjoianJvY2tldEBleGFtcGxlLmNvbSJ9.4O-W8aeOOUFU8fkGTAFaAu9D9nK_l61yPomNG1Qkny8',
  orgs: {
    danlaw: {
      id: 'bd006616-3bd2-4783-a84d-b0fa4ed81135',
      properName: 'Danlaw',
    },
    azuga: {
      id: '8d9ec637-f3bb-4c46-83ee-11764d09f0b2',
      properName: 'Azuga',
    },
    bitbrew: {
      id: '94d9b0ca-a758-4502-8dcd-41d883e38714',
      properName: 'BitBrew',
    },
  },
  users: {
    bill: {
      email: 'bill@bitbrew.com',
      name: 'Bill Monaghan',
    },
    cynthia: {
      email: 'cynthia@bitbrew.com',
      name: 'Cynthia Hamilton',
    },
    alex: {
      email: 'bill@bitbrew.com',
      name: 'Alex Li',
    },
  },
};
