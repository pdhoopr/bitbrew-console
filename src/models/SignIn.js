import { getEnv, getSnapshot, types } from 'mobx-state-tree';

export default types
  .model('SignIn', {
    accessToken: types.optional(types.string, ''),
  })
  .actions(self => ({
    update(event) {
      self.accessToken = event.currentTarget.value;
    },
    submit(event) {
      event.preventDefault();
      getEnv(self).onSubmit(getSnapshot(self));
    },
  }));
