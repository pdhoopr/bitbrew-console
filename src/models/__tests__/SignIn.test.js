import { fixtures } from 'bitbrew-test-helpers';
import { getSnapshot } from 'mobx-state-tree';
import SignIn from '../SignIn';

test('initializes sign in with the default values', () => {
  const model = SignIn.create();
  expect(getSnapshot(model)).toMatchSnapshot();
});

test('initializes sign in with the defined values', () => {
  const data = {
    accessToken: fixtures.accessToken,
  };
  const model = SignIn.create(data);
  expect(model).toMatchObject(data);
});

test('updates the value of the access token', () => {
  const token = fixtures.accessToken;
  const model = SignIn.create();
  expect(model.accessToken).not.toBe(token);
  model.update({
    currentTarget: {
      id: 'accessToken',
      value: token,
    },
  });
  expect(model.accessToken).toBe(token);
});

test('submits sign in data to the callback defined in the environment', () => {
  const env = {
    onSubmit: jest.fn(),
  };
  const event = {
    preventDefault: jest.fn(),
  };
  const model = SignIn.create({}, env);
  model.update({
    currentTarget: {
      id: 'accessToken',
      value: fixtures.accessToken,
    },
  });
  model.submit(event);
  expect(event.preventDefault).toHaveBeenCalled();
  expect(env.onSubmit).toHaveBeenCalledWith(getSnapshot(model));
});
