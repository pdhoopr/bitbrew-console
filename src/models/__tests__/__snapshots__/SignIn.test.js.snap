// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initializes sign in with the default values 1`] = `
Object {
  "accessToken": "",
}
`;
